include/slang/binding/AssignmentExpressions.h

```diff
@@ -65,15 +65,19 @@ class AssignmentExpression : public Expression {
 struct CastExpressionSyntax;
 struct SignedCastExpressionSyntax;
 
+enum class ConvertCast : uint8_t { Propagate, Implicit, BitstreamCast, Explicit };
+
 /// Represents a type conversion expression (implicit or explicit).
 class ConversionExpression : public Expression {
 public:
-    bool isImplicit;
+    bool isImplicit() const { return cast_ <= ConvertCast::Implicit; }
+    void setCast(ConvertCast);
```

Comment:
You can make the conversionKind a free member, no need to wrap it in simple getters / setters.

Reply:
I guess you meant to make this member "public" and other classes can freely modify it directly. I can make the change.

Reply:
I make it a public const member.
```c++
public:
    const ConversionKind castKind;
```
which cannot be modified after construction.